### ISSUE TITLE
Rust file stuck on "Analyzing"

### ISSUE THREAD

**Issue by kylepotts** [intellij-rust/intellij-rust#6235]
## Environment

* **IntelliJ Rust plugin version:**
Version 0.3.132.3395-203
* **Rust toolchain version:**
1.46.0
* **IDE name and version:**
IntelliJ IDEA 2020.3 EAP (Community Edition)
Build #IC-203.4449.2, built on October 6, 2020
Runtime version: 11.0.8+10-b1098.1 amd64
Current Desktop: Pantheon
* **Operating system:**
Linux
## Problem description

I do not get any Intellijsense on my rust file. Code analysis is stuck on "Analyzing"

## Steps to reproduce

Open or create any rust project with the rust plug in installed

```
2020-10-10 08:38:11,423 [  10125]  ERROR -                org.rust.macros - Error during macro expansion 
java.lang.NoSuchMethodError: 'void com.intellij.util.containers.ConcurrentWeakKeySoftValueHashMap.<init>(int, float, int, gnu.trove.TObjectHashingStrategy)'
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt$createWeakMap$1.<init>(RsResolveCache.kt:270)
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt.createWeakMap(RsResolveCache.kt:274)
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt.getOrCreateMap(RsResolveCache.kt:264)
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt.access$getOrCreateMap(RsResolveCache.kt:1)
	at org.rust.lang.core.resolve.ref.RsResolveCache.getRustStructureDependentCache(RsResolveCache.kt:57)
	at org.rust.lang.core.resolve.ref.RsResolveCache.getCacheFor(RsResolveCache.kt:151)
	at org.rust.lang.core.resolve.ref.RsResolveCache.resolveWithCaching(RsResolveCache.kt:97)
	at org.rust.lang.core.resolve.ref.RsReferenceCached.cachedMultiResolve(RsReferenceCached.kt:27)
	at org.rust.lang.core.resolve.ref.RsReferenceCached.multiResolve(RsReferenceCached.kt:20)
	at com.intellij.psi.PsiPolyVariantReferenceBase.resolve(PsiPolyVariantReferenceBase.java:47)
	at org.rust.lang.core.resolve.ref.RsReferenceBase.resolve(RsReferenceBase.kt:28)
	at org.rust.lang.core.stubs.index.RsModulesIndex$Companion$getDeclarationsFor$1$1.process(RsModulesIndex.kt:46)
	at org.rust.lang.core.stubs.index.RsModulesIndex$Companion$getDeclarationsFor$1$1.process(RsModulesIndex.kt:31)
	at com.intellij.psi.stubs.StubProcessingHelperBase.processStubsInFile(StubProcessingHelperBase.java:73)
	at com.intellij.psi.stubs.StubIndexImpl.lambda$processElements$2(StubIndexImpl.java:289)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:325)
	at com.intellij.psi.stubs.StubIndex.processElements(StubIndex.java:49)
	at org.rust.lang.core.stubs.index.RsModulesIndex$Companion$getDeclarationsFor$1.invoke(RsModulesIndex.kt:43)
	at org.rust.lang.core.stubs.index.RsModulesIndex$Companion.getDeclarationsFor(RsModulesIndex.kt:52)
	at org.rust.lang.core.psi.RsFile$declarations$1.compute(RsFile.kt:206)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:153)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:54)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$1(CachedValueBase.java:235)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:42)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:236)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:43)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:76)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:150)
	at org.rust.lang.core.psi.RsFile.getDeclarations(RsFile.kt:205)
	at org.rust.lang.core.psi.RsFile.getDeclaration(RsFile.kt:190)
	at org.rust.lang.core.psi.RsFile.doGetCachedData(RsFile.kt:101)
	at org.rust.lang.core.psi.RsFile.access$doGetCachedData(RsFile.kt:68)
	at org.rust.lang.core.psi.RsFile$cachedData$1$value$1.compute(RsFile.kt:93)
	at org.rust.lang.core.psi.RsFile$cachedData$1$value$1.compute(RsFile.kt:68)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:42)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at org.rust.openapiext.UtilsKt.recursionGuard(utils.kt:69)
	at org.rust.openapiext.UtilsKt.recursionGuard$default(utils.kt:68)
	at org.rust.lang.core.psi.RsFile$cachedData$1.compute(RsFile.kt:93)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:153)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:54)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$1(CachedValueBase.java:235)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:42)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:236)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:43)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:76)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:150)
	at org.rust.lang.core.psi.RsFile.getCachedData(RsFile.kt:92)
	at org.rust.lang.core.psi.RsFile.getCrateRoot(RsFile.kt:77)
	at org.rust.lang.core.macros.SourceFile.extract(ExpandedMacroStorage.kt:533)
	at org.rust.lang.core.macros.SourceFile.extract(ExpandedMacroStorage.kt:498)
	at org.rust.lang.core.macros.Extractable.extract(MacroExpansionTask.kt:328)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1$stages1$1$result$1$1.invoke(MacroExpansionTask.kt:190)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1$stages1$1$result$1$1.invoke(MacroExpansionTask.kt:48)
	at org.rust.openapiext.UtilsKt$runReadActionInSmartMode$1.compute(utils.kt:275)
	at com.intellij.openapi.project.DumbService.lambda$runReadActionInSmartMode$0(DumbService.java:103)
	at com.intellij.openapi.project.DumbService.lambda$runReadActionInSmartMode$1(DumbService.java:147)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:890)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.openapi.project.DumbService.runReadActionInSmartMode(DumbService.java:140)
	at com.intellij.openapi.project.DumbService.runReadActionInSmartMode(DumbService.java:103)
	at org.rust.openapiext.UtilsKt.runReadActionInSmartMode(utils.kt:273)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1$stages1$1$result$1.invoke(MacroExpansionTask.kt:187)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1$stages1$1$result$1.invoke(MacroExpansionTask.kt:48)
	at org.rust.openapiext.UtilsKt$executeUnderProgressWithWriteActionPriorityWithRetries$success$1.invoke(utils.kt:285)
	at org.rust.openapiext.UtilsKt$executeUnderProgressWithWriteActionPriorityWithRetries$success$1.invoke(utils.kt)
	at org.rust.openapiext.UtilsKt$sam$java_lang_Runnable$0.run(utils.kt)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:152)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:113)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:63)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:59)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:110)
	at org.rust.openapiext.UtilsKt.runWithWriteActionPriority(utils.kt:297)
	at org.rust.openapiext.UtilsKt.executeUnderProgressWithWriteActionPriorityWithRetries(utils.kt:284)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1$stages1$1.apply(MacroExpansionTask.kt:185)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1$stages1$1.apply(MacroExpansionTask.kt:48)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:952)
	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:926)
	at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:327)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:408)
	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:736)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:919)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at kotlin.streams.jdk8.StreamsKt.toList(Streams.kt:60)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1.invoke(MacroExpansionTask.kt:201)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1.invoke(MacroExpansionTask.kt:48)
	at org.rust.stdext.CompletableFutureKt$supplyAsync$1.get(CompletableFuture.kt:18)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
2020-10-10 08:38:11,425 [  10127]  ERROR -                org.rust.macros - IntelliJ IDEA 2020.3 EAP  Build #IC-203.4449.2 
2020-10-10 08:38:11,426 [  10128]  ERROR -                org.rust.macros - JDK: 11.0.8; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o. 
2020-10-10 08:38:11,426 [  10128]  ERROR -                org.rust.macros - OS: Linux 
2020-10-10 08:38:11,426 [  10128]  ERROR -                org.rust.macros - Plugin to blame: Rust version: 0.3.132.3395-203 
2020-10-10 08:38:11,426 [  10128]  ERROR -                org.rust.macros - Last Action:  
2020-10-10 08:38:11,808 [  10510]  ERROR - aemon.impl.PassExecutorService - 'void com.intellij.util.containers.ConcurrentWeakKeySoftValueHashMap.<init>(int, float, int, gnu.trove.TObjectHashingStrategy)' 
java.lang.NoSuchMethodError: 'void com.intellij.util.containers.ConcurrentWeakKeySoftValueHashMap.<init>(int, float, int, gnu.trove.TObjectHashingStrategy)'
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt$createWeakMap$1.<init>(RsResolveCache.kt:270)
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt.createWeakMap(RsResolveCache.kt:274)
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt.access$createWeakMap(RsResolveCache.kt:1)
	at org.rust.lang.core.resolve.ref.RsResolveCache$getCacheFor$2.compute(RsResolveCache.kt:143)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:153)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:54)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$1(CachedValueBase.java:235)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:42)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:236)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:43)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:76)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:150)
	at org.rust.lang.core.resolve.ref.RsResolveCache.getCacheFor(RsResolveCache.kt:142)
	at org.rust.lang.core.resolve.ref.RsResolveCache.resolveWithCaching(RsResolveCache.kt:97)
	at org.rust.lang.core.resolve.ref.RsPathReferenceImpl.advancedCachedMultiResolve(RsPathReferenceImpl.kt:89)
	at org.rust.lang.core.resolve.ref.RsPathReferenceImpl.advancedMultiResolve(RsPathReferenceImpl.kt:85)
	at org.rust.lang.core.resolve.ref.RsPathReferenceImpl.advancedResolve(RsPathReferenceImpl.kt:74)
	at org.rust.lang.core.types.infer.DeclarationsKt.inferTypeReferenceType(Declarations.kt:38)
	at org.rust.lang.core.types.infer.DeclarationsKt.inferTypeReferenceType$default(Declarations.kt:25)
	at org.rust.lang.core.types.ExtensionsKt.getType(Extensions.kt:53)
	at org.rust.lang.core.psi.ext.RsFunctionKt.getReturnType(RsFunction.kt:102)
	at org.rust.lang.core.types.infer.RsInferenceContext.infer(TypeInference.kt:167)
	at org.rust.lang.core.types.infer.TypeInferenceKt$inferTypesIn$1.compute(TypeInference.kt:34)
	at org.rust.lang.core.types.infer.TypeInferenceKt$inferTypesIn$1.compute(TypeInference.kt)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:42)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at org.rust.openapiext.UtilsKt.recursionGuard(utils.kt:69)
	at org.rust.lang.core.types.infer.TypeInferenceKt.inferTypesIn(TypeInference.kt:34)
	at org.rust.lang.core.types.ExtensionsKt$inference$1.compute(Extensions.kt:69)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:153)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:54)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$1(CachedValueBase.java:235)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:42)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:236)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:43)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:76)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:150)
	at org.rust.lang.core.types.ExtensionsKt.getInference(Extensions.kt:68)
	at org.rust.lang.core.types.ExtensionsKt.getInference(Extensions.kt:159)
	at org.rust.lang.core.resolve.ref.RsPathReferenceImpl.advancedMultiResolve(RsPathReferenceImpl.kt:84)
	at org.rust.lang.core.resolve.ref.RsPathReferenceImpl.multiResolve(RsPathReferenceImpl.kt:77)
	at com.intellij.psi.PsiPolyVariantReferenceBase.resolve(PsiPolyVariantReferenceBase.java:47)
	at org.rust.lang.core.resolve.ref.RsReferenceBase.resolve(RsReferenceBase.kt:28)
	at org.rust.ide.utils.CallInfo$Companion.resolve(CallInfo.kt:26)
	at org.rust.ide.hints.parameter.RsInlayParameterHints.provideHints(RsInlayParameterHints.kt:34)
	at org.rust.ide.hints.parameter.RsInlayParameterHintsProvider.getParameterHints(RsInlayParameterHintsProvider.kt:40)
	at com.intellij.codeInsight.hints.InlayParameterHintsProvider.getParameterHints(InlayParameterHintsProvider.java:29)
	at com.intellij.codeInsight.hints.ParameterHintsPass.process(ParameterHintsPass.java:74)
	at com.intellij.codeInsight.hints.ParameterHintsPass.lambda$doCollectInformation$0(ParameterHintsPass.java:66)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at com.intellij.codeInsight.hints.ParameterHintsPass.doCollectInformation(ParameterHintsPass.java:66)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:54)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:401)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1138)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:394)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:63)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:393)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:369)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:172)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:183)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:367)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:188)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
2020-10-10 08:38:11,809 [  10511]  ERROR - aemon.impl.PassExecutorService - IntelliJ IDEA 2020.3 EAP  Build #IC-203.4449.2 
2020-10-10 08:38:11,809 [  10511]  ERROR - aemon.impl.PassExecutorService - JDK: 11.0.8; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o. 
2020-10-10 08:38:11,809 [  10511]  ERROR - aemon.impl.PassExecutorService - OS: Linux 
2020-10-10 08:38:11,809 [  10511]  ERROR - aemon.impl.PassExecutorService - Plugin to blame: Rust version: 0.3.132.3395-203 
2020-10-10 08:38:11,809 [  10511]  ERROR - aemon.impl.PassExecutorService - Last Action:  
2020-10-10 08:38:11,819 [  10521]  ERROR - aemon.impl.PassExecutorService - 'void com.intellij.util.containers.ConcurrentWeakKeySoftValueHashMap.<init>(int, float, int, gnu.trove.TObjectHashingStrategy)' 
java.lang.NoSuchMethodError: 'void com.intellij.util.containers.ConcurrentWeakKeySoftValueHashMap.<init>(int, float, int, gnu.trove.TObjectHashingStrategy)'
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt$createWeakMap$1.<init>(RsResolveCache.kt:270)
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt.createWeakMap(RsResolveCache.kt:274)
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt.access$createWeakMap(RsResolveCache.kt:1)
	at org.rust.lang.core.resolve.ref.RsResolveCache$getCacheFor$2.compute(RsResolveCache.kt:143)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:153)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:54)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$1(CachedValueBase.java:235)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:42)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:236)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:43)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:76)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:150)
	at org.rust.lang.core.resolve.ref.RsResolveCache.getCacheFor(RsResolveCache.kt:142)
	at org.rust.lang.core.resolve.ref.RsResolveCache.resolveWithCaching(RsResolveCache.kt:97)
	at org.rust.lang.core.resolve.ref.RsPathReferenceImpl.advancedCachedMultiResolve(RsPathReferenceImpl.kt:89)
	at org.rust.lang.core.resolve.ref.RsPathReferenceImpl.advancedMultiResolve(RsPathReferenceImpl.kt:85)
	at org.rust.lang.core.resolve.ref.RsPathReferenceImpl.advancedResolve(RsPathReferenceImpl.kt:74)
	at org.rust.lang.core.types.infer.DeclarationsKt.inferTypeReferenceType(Declarations.kt:38)
	at org.rust.lang.core.types.infer.DeclarationsKt.inferTypeReferenceType$default(Declarations.kt:25)
	at org.rust.lang.core.types.ExtensionsKt.getType(Extensions.kt:53)
	at org.rust.lang.core.psi.ext.RsFunctionKt.getReturnType(RsFunction.kt:102)
	at org.rust.lang.core.types.infer.RsInferenceContext.infer(TypeInference.kt:167)
	at org.rust.lang.core.types.infer.TypeInferenceKt$inferTypesIn$1.compute(TypeInference.kt:34)
	at org.rust.lang.core.types.infer.TypeInferenceKt$inferTypesIn$1.compute(TypeInference.kt)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:42)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at org.rust.openapiext.UtilsKt.recursionGuard(utils.kt:69)
	at org.rust.lang.core.types.infer.TypeInferenceKt.inferTypesIn(TypeInference.kt:34)
	at org.rust.lang.core.types.ExtensionsKt$inference$1.compute(Extensions.kt:69)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:153)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:54)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$1(CachedValueBase.java:235)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:42)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:236)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:43)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:76)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:150)
	at org.rust.lang.core.types.ExtensionsKt.getInference(Extensions.kt:68)
	at org.rust.lang.core.types.ExtensionsKt.getInference(Extensions.kt:159)
	at org.rust.lang.core.resolve.ref.RsPathReferenceImpl.advancedMultiResolve(RsPathReferenceImpl.kt:84)
	at org.rust.lang.core.resolve.ref.RsPathReferenceImpl.multiResolve(RsPathReferenceImpl.kt:77)
	at com.intellij.psi.PsiPolyVariantReferenceBase.resolve(PsiPolyVariantReferenceBase.java:47)
	at org.rust.lang.core.resolve.ref.RsReferenceBase.resolve(RsReferenceBase.kt:28)
	at org.rust.ide.utils.CallInfo$Companion.resolve(CallInfo.kt:26)
	at org.rust.ide.hints.parameter.RsInlayParameterHints.provideHints(RsInlayParameterHints.kt:34)
	at org.rust.ide.hints.parameter.RsInlayParameterHintsProvider.getParameterHints(RsInlayParameterHintsProvider.kt:40)
	at com.intellij.codeInsight.hints.InlayParameterHintsProvider.getParameterHints(InlayParameterHintsProvider.java:29)
	at com.intellij.codeInsight.hints.ParameterHintsPass.process(ParameterHintsPass.java:74)
	at com.intellij.codeInsight.hints.ParameterHintsPass.lambda$doCollectInformation$0(ParameterHintsPass.java:66)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at com.intellij.codeInsight.hints.ParameterHintsPass.doCollectInformation(ParameterHintsPass.java:66)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:54)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:401)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1138)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:394)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:63)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:393)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:369)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:172)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:183)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:367)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:188)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
2020-10-10 08:38:11,821 [  10523]  ERROR - aemon.impl.PassExecutorService - IntelliJ IDEA 2020.3 EAP  Build #IC-203.4449.2 
2020-10-10 08:38:11,824 [  10526]  ERROR - aemon.impl.PassExecutorService - JDK: 11.0.8; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o. 
2020-10-10 08:38:11,824 [  10526]  ERROR - aemon.impl.PassExecutorService - OS: Linux 
2020-10-10 08:38:11,824 [  10526]  ERROR - aemon.impl.PassExecutorService - Plugin to blame: Rust version: 0.3.132.3395-203 
2020-10-10 08:38:11,824 [  10526]  ERROR - aemon.impl.PassExecutorService - Last Action:  

```

**Comment by Undin** [comment]
Duplicate of #6225
Already fixed in #6223
You can try install nightly builds. See [instruction](https://intellij-rust.github.io/install#nightly-builds) how to do it

**Comment by kylepotts** [comment]
@Undin Thanks I ended up downgrading to a non EAP version Glad its being fixed. Thanks!
